Lib/importlib/_bootstrap_external.py

```diff
@@ -140,13 +157,14 @@ def _path_isdir(path):
     return _path_is_mode_type(path, 0o040000)
 
 
-def _path_isabs(path):
-    """Replacement for os.path.isabs.
-
-    Considers a Windows drive-relative path (no drive, but starts with slash) to
-    still be "absolute".
-    """
-    return path.startswith(path_separators) or path[1:3] in _pathseps_with_colon
+if _MS_WINDOWS:
+    def _path_isabs(path):
+        """Replacement for os.path.isabs."""
+        return path and bool(_os._path_splitroot(path)[0])
+else:
```

Comment:
You have lost me here. This will return true for drive-relative paths such as "C:spam" and rooted paths such as "\\spam". In Windows, neither one of these is an absolute path. They're relative to the drive of the current working directory or per-drive working directory. The DLL loader will not allow them in the DLL search path. So `LoadLibraryExW` will fail if the flag `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` is used and the DLL has dependencies that have to be search for, i.e. that aren't already loaded, known DLLs, or API sets.

Reply:
BTW, it's not the fault of `PathCchSkipRoot`. It's working as advertised. It doesn't claim to determine whether a path is relative or absolute, only to skip the drive root if the path includes it. "C:spam" expands to "C:\\drive\\working\\directory\\spam". Whatever the working directory happens to be, it must include the root path. The only way to skip it is to skip all of "C:", whatever that path is. The alternative is not skipping the root, which wouldn't be better. That doesn't mean that this is an absolute path that can be relied on. At any time, another thread could change the working directory on drive "C:", which could lead to disastrously inconsistent results for anyone who mistakenly handles "C:spam" as an absolute path.

Reply:
Yeah, still finding gaps in here, those are resolvable though. Got stuck for two hours because someone changed the bytecode without changing the pyc magic number, and I didn't go through and wipe all my existing files...